src/box/txn_limbo.c

```diff
@@ -235,6 +237,8 @@ txn_limbo_wait_complete(struct txn_limbo *limbo, struct txn_limbo_entry *entry)
 		double deadline = start_time + replication_synchro_timeout;
 		double timeout = deadline - fiber_clock();
 		int rc = fiber_cond_wait_timeout(&limbo->wait_cond, timeout);
+		if (limbo->frozen)
+			goto wait;
```

Comment:
What if I disable fencing box.cfg option while having a transaction hanging in the infinite waiting state? Shouldn't it continue waiting for a deadline then and write ROLLBACK eventually?

Reply:
I tried to keep limbo freezing as simple as possible. Right now there is a simple rule on when it unfreezes: "When PROMOTE/DEMOTE is read for a new term". As stated in #6661:
`Additionally the former leader shouldn't CONFIRM/ROLLBACK any of the synchronous transactions which were in flight at the moment of step down. Otherwise we might end up in a situation when the leader which lost connectivity rolls back a transaction which is already confirmed by a new leader.`
By unfreezing limbo on box.cfg.ellection_fencing_enabled we might end up sending ROLLBACKs to still connected replicas to the old leader, while there might be a new one we don't see right now and he already confirmed those transactions.